lib/ash/code_interface.ex

```diff
@@ -564,7 +564,9 @@ defmodule Ash.CodeInterface do
               resolve_subject =
                 quote do
                   {query_opts, opts} =
-                    Keyword.split(opts, [:query, :actor, :tenant, :authorize?, :tracer])
+                    Keyword.split(opts, [:query, :actor, :tenant, :authorize?, :tracer, :context])
+
+                  {context, query_opts} = Keyword.pop(query_opts, :context, %{})
```

Comment:
for_x opts do not take a context, but it would make sense to set the context before passing it to for_x

Reply:
Lets make them accept a context instead.

Reply:
> for_x opts do not take a context, but it would make sense to set the context before passing it to for_x

https://github.com/ash-project/ash/pull/1016#issuecomment-2051709371